libr/include/r_anal.h

```diff
@@ -215,6 +215,33 @@ enum {
 	R_ANAL_DIFF_TYPE_UNMATCH = 'u'
 };
 
+typedef struct enum_case {
+	const char *name;
+	const int val;
+} RAnalEnumCase;
+
+typedef enum {
+	R_ANAL_BASE_TYPE_KIND_STRUCT,
+	R_ANAL_BASE_TYPE_KIND_UNION,
+	R_ANAL_BASE_TYPE_KIND_ENUM,
+} RAnalBaseTypeKind;
+
+typedef struct r_anal_base_type_struct {
+	RVector *members; // describe the struct in some way
```

Comment:
Any particular reason you made it `RVector *`? It's just a very minor optimization, but it's usually better to not have a pointer and use init/fini.

Reply:
Just for an ease of use, because all the functions worked with `RVector *`, changed it back to a non pointer